pkg/costmodel/allocation_helpers.go

```diff
@@ -642,15 +650,66 @@ func applyGPUUsageAvg(podMap map[podKey]*pod, resGPUUsageAvg []*prom.QueryResult
 		for _, thisPod := range pods {
 			container, err := res.GetString("container")
 			if err != nil {
-				log.DedupedWarningf(10, "CostModel.ComputeAllocation: GPU usage avg query result missing 'container': %s", key)
+				log.DedupedWarningf(10, "CostModel.ComputeAllocation: GPU usage avg/max query result missing 'container': %s", key)
 				continue
 			}
 			if _, ok := thisPod.Allocations[container]; !ok {
 				thisPod.appendContainer(container)
 			}
 
 			// DCGM_FI_PROF_GR_ENGINE_ACTIVE metric is a float between 0-1.
-			thisPod.Allocations[container].GPUUsageAverage = res.Values[0].Value
+			switch mode {
+			case GpuUsageAverageMode:
+
+				if thisPod.Allocations[container].GPUAllocation == nil {
+					thisPod.Allocations[container].GPUAllocation = &opencost.GPUAllocation{GPUUsageAverage: &res.Values[0].Value}
+				} else {
+					thisPod.Allocations[container].GPUAllocation.GPUUsageAverage = &res.Values[0].Value
+				}
+			case GpuUsageMaxMode:
+				thisPod.Allocations[container].RawAllocationOnly.GPUUsageMax = &res.Values[0].Value
+			case GpuIsSharedMode:
+				// if a container is using a GPU and it is shared, isGPUShared will be true
+				// if a container is using GPU and it is NOT shared, isGPUShared will be false
+				// if a container is NOT using a GPU, isGPUShared will be null
+				if res.Metric["resource"] == "nvidia_com_gpu_shared" {
```

Comment:
@chipzoller Just want to double check with you. Is this the primary way or only way you can request a shared (specifically time spliced) GPU?

```
resources:
  limits:
    nvidia.com/gpu.shared: "1" 
  requests:
    nvidia.com/gpu.shared: "1" 
```

Reply:
This isn't technically the only way a shared GPU can be requested, but it is the only way to disambiguate a shared versus non-shared GPU request when looking at the Pod spec itself. Yes, it represents an opinionated approach we want to take (to start), but it's predictable and simpler. The sharing method cannot be specified via the static device plugin today. This is a gap that will be addressed by Dynamic Resource Allocation (DRA) in future versions of Kubernetes.

Reply:
Ok great! Thanks for clarifying.